manager-bundle/src/ContaoManager/Plugin.php

```diff
@@ -369,6 +370,77 @@ private function addDefaultPdoDriverOptions(array $extensionConfigs): array
         return $extensionConfigs;
     }
 
+    /**
+     * Adds a default orm mapping for the App namespace if none was configured.
+     *
+     * @return array<string,array<string,array<string,array<string,mixed>>>>
+     */
+    private function addDefaultDoctrineMapping(array $extensionConfigs, ContainerBuilder $container): array
+    {
+        $defaultEntityManager = 'default';
+
+        foreach ($extensionConfigs as $config) {
+            if (null !== $em = $config['orm']['default_entity_manager'] ?? null) {
+                $defaultEntityManager = $em;
+            }
+        }
+
+        $mappings = [];
+        $autoMappingEnabled = false;
+
+        foreach ($extensionConfigs as $config) {
+            $mappings[] = $config['orm']['mappings'] ?? [];
+
+            foreach ($config['orm']['entity_managers'] ?? [] as $em) {
+                $mappings[] = $em['mappings'] ?? [];
+            }
+
+            $autoMappingEnabled |= ($config['orm']['auto_mapping'] ?? false) ||
+                ($config['orm']['entity_managers'][$defaultEntityManager]['auto_mapping'] ?? false);
+        }
+
+        // Skip if auto mapping isn't enabled for the default entity manager.
+        if (!$autoMappingEnabled) {
+            return $extensionConfigs;
+        }
+
+        // Skip if a mapping with name or alias 'App' already exists or any
+        // mapping already targets '%kernel.project_dir%/src/Entity'.
+        foreach (array_replace(...$mappings) as $name => $values) {
+            if (
+                'App' === $name || 'App' === ($values['alias'] ?? '') ||
+                '%kernel.project_dir%/src/Entity' === ($values['dir'] ?? '')
+            ) {
+                return $extensionConfigs;
+            }
+        }
+
+        // Skip if the '%kernel.project_dir%/src/Entity' directory does not exists.
```

Comment:
we might wanna do this before merging and validating over the configs as it seems less work.

Reply:
:thinking: I'd assume the file access to be much slower than the array shenanigans.